pkgs/pkgs-lib/formats.nix

```
@@ -1125,4 +1125,34 @@ optionalAttrs allowAliases aliases
     else
       throw "pkgs.formats.xml: Unknown format: ${format}";
 
+  plist =
+    {
+      escape ? true,
+    }:
+    {
+      type =
+        let
+          valueType =
+            nullOr (oneOf [
+              bool
+              int
+              float
+              str
+              path
+              (attrsOf valueType)
+              (listOf valueType)
+            ])
+            // {
+              description = "Proper list value";
+            };
+        in
+        valueType;
+
+      generate =
+        name: value:
+        pipe value [
+          (lib.generators.toPlist { inherit escape; })
+          (pkgs.writeText name)
+        ];
```

I don't really think the pipe is necessary here. It's certainly nice to have for complicated transformations, but tends to create very ugly stacktraces.

```suggestion
        pkgs.writeText name (lib.generators.toPlist { inherit escape; } value);
```

Alternatively

```suggestion
        let
          content = lib.generators.toPlist { inherit escape; } value;
        in
          pkgs.writeText name content;
```